pkg/cmd/cli/secrets/new.go

```diff
@@ -144,18 +147,22 @@ func (o *CreateSecretOptions) Validate() error {
 		return errors.New("at least one source file or directory must be specified")
 	}
 
-nameCheck:
-	switch o.SecretTypeName {
-	case string(kapi.SecretTypeOpaque), "":
-		// this is ok
-	default:
-		// TODO this probably isn't a good idea.  It limits the power of this command.  Maybe allow unknown names with a force?
-		for _, secretType := range KnownSecretTypes {
-			if o.SecretTypeName == string(secretType.Type) {
-				break nameCheck
+	if !o.AllowUnknownTypes {
```

Comment:
Should either validate that `--confirm` can only be used when `--type=<type>` was provided, or move this line a little below, inside the `default:` section of the switch. To make sure we test secret type name when `--confirm` were provided without `--type` (I know it does nothing today, but still).